src/primitives/cone.jl

```diff
@@ -0,0 +1,30 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  Cone(vertex, center, radius, nsides, closed)
+
+Cone with `vertex` and base with `center` and `radius` made of `nsides` 
+segments. If `closed`, the base is included.
+"""
+struct Cone{T} <: Primitive{3,T}
+  vertex::Point{3,T}
+  center::Point{3,T}
+  radius::T
+  nsides::Int
+  closed::Bool
```

Comment:
The definition of a Cone relies on an apex and a plane. We can assume that the base of the cone is a perfect circle, but it could be any simple geometry like a polygon: https://en.wikipedia.org/wiki/Cone

A cone with polygonal base is called a pyramid. We have a Pyramid type already but we assume a quadrangle base currently. We should probably generalize it to accept any polygon.